vpr/src/base/partition.h

```diff
@@ -0,0 +1,27 @@
+#ifndef PARTITION_H
+#define PARTITION_H
+
+#include "vtr_strong_id.h"
+#include "region.h"
+
+/**
+ * This class stores the data for each constraint partition -
+ * partition name, the atoms that are in the partition,
+ * and the regions that the partition can be placed in.
+ */
+
+//Type tag for PartitionId
+struct partition_id_tag;
+
+//A unique identifier for a partition
+typedef vtr::StrongId<partition_id_tag> PartitionId;
+
+class Partition {
+  public: //accessors
+  private:
+    PartitionId id;
+    std::string name;
+    std::vector<AtomBlockId> atomBlocks; //atoms that belong to this partition
+    std::vector<Region> regions;         //regions that this partition can be placed in
```

Comment:
if multiple regions, legal to place in any of them (i.e. union).

Reply:
Should use PartitionRegions type here -- that's why you made a class.  Decide on best place to comment fact that a PartitionRegions is the union of multiple regions (should be in the PartitionRegions class mostly, can have brief comments outside).